server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -187,15 +188,36 @@ protected void closeInternal() {
     };
 
     public Store(ShardId shardId, IndexSettings indexSettings, Directory directory, ShardLock shardLock) {
-        this(shardId, indexSettings, directory, shardLock, OnClose.EMPTY);
+        this(shardId, indexSettings, directory, null, shardLock, OnClose.EMPTY);
+    }
+
+    public Store(ShardId shardId, IndexSettings indexSettings, Directory directory, Directory secondaryDirectory, ShardLock shardLock) {
```

Comment:
2 extra constructor methods due to `secondaryDirectory`. Should we use a builder instead?

Reply:
> Should we use a builder instead?

Yes, and the builder should validate the arguments and throw a `RuntimeException` if the `REMOTE_STORE` feature flag is not set.

Reply:
In the latest commit, I have removed extra parameter to `Store`. So, this is not required anymore.